pytorch_lightning/profiler/profilers.py

```diff
@@ -88,15 +117,41 @@ def profile_iterable(self, iterable, action_name: str) -> None:
 
     def describe(self) -> None:
         """Logs a profile report after the conclusion of the training run."""
+        self._prepare_file()
         for write in self.write_streams:
             write(self.summary())
+        if self.output_file:
+            self.output_file.flush()
+        self.teardown()
+
+    def stats_to_str(self, stats: Dict[str, str]) -> str:
+        output = ["Profiler Report"]
+        for action, value in stats.items():
+            header = f"Profile stats for: {action}"
+            if getattr(self, "local_rank", None) is not None:
+                header += f" rank: {self.local_rank}"
+            output.append(header)
+            output.append(value)
+        return os.linesep.join(output)
+
+    def teardown(self) -> None:
+        """Close profiler's stream."""
+        if self.output_file:
+            self.output_file.close()
+        self.write_streams = []
+        self._file_prepared = False
 
-    @abstractmethod
-    def summary(self) -> str:
-        """Create profiler summary in text format."""
+    def __del__(self) -> None:
+        self.teardown()
 
-    def on_train_start(self, local_rank: Optional[int] = None):
-        self.local_rank = local_rank
+    def start(self, action_name: str) -> None:
+        raise NotImplementedError
+
+    def stop(self, action_name: str) -> None:
+        raise NotImplementedError
+
+    def summary(self) -> str:
+        raise NotImplementedError
```

Comment:
n00b q: why are these needed? would someone directly instantiate the `BaseProfiler`?

Reply:
To avoid the IDE warning `Class BaseProfiler must implement all abstract methods `